### ISSUE TITLE
app crash: Cannot read property 'users' of undefined

### ISSUE THREAD

**Issue by peerbolte** [mrvautin/adminMongo#1]
First of all, thanks for creating this application. It's exactly what I was looking for. However I run into an error. When connecting to my local dev database, everything is OK. However when I try to connect to an external database, which is created on my VPS using dokku, the app crashes with the following error:
```
/adminMongo/node_modules/mongodb/lib/utils.js:98
    process.nextTick(function() { throw err; });
                                  ^

TypeError: Cannot read property 'users' of undefined
    at /adminMongo/routes/index.js:56:55
    at get_db_list (/adminMongo/routes/index.js:864:9)
    at /adminMongo/routes/index.js:51:25
    at /adminMongo/routes/index.js:902:13
    at /adminMongo/node_modules/mongojs/lib/database.js:121:5
    at /adminMongo/node_modules/mongojs/lib/database.js:113:7
    at handleCallback (/adminMongo/node_modules/mongodb/lib/utils.js:96:12)
    at /adminMongo/node_modules/mongodb/lib/cursor.js:851:16
    at handleCallback (/adminMongo/node_modules/mongodb-core/lib/cursor.js:244:5)
    at setCursorDeadAndNotified (/adminMongo/node_modules/mongodb-core/lib/cursor.js:553:3)
```

when I use the following connection string in terminal using mongo shell, I do get a connection:
```
mongo <IP>:<PORT>/<DB> -u <USER> -p <PASS>
```

I'm using the following connection string in adminMongo
```
mongodb://<USER>:<PASS>@<IP>:<PORT>/<DB>
```

The database version I'm connecting to is 3.0.6

Any ideas what might go wrong? Thanks in advance

**Comment by mrvautin** [comment]
Thanks for the positive feedback and the bug report. The issue looks to be user permissions related. adminMongo is trying to run "{ usersInfo: 1 }" to get a list of all users. Are you able to run this form the shell?

Regardless, I have a feeling I will need to make a change for user accounts without permissions to list all users of a given DB.

**Comment by peerbolte** [comment]
Thanks for the quick reply. When I run the command from shell:

```
db.runCommand({usersInfo:1})
{
	"ok" : 0,
	"errmsg" : "not authorized on XXX to execute command { usersInfo: 1.0 }",
	"code" : 13
}
```
So I think it's indeed a permission issue.

**Comment by mrvautin** [comment]
Great. Are you able to run db.getUser("userid") from shell to determine what user roles are assigned to your account? I'm trying to replicate the permissions.

**Comment by peerbolte** [comment]
If I replace userid with the actual user id specified in the connection string, I will get a response in the following format:
```
db.getUser("<USER>")
```
returns
```
"_id" : "<DB.USER >",
	"user" : "< USER >",
	"db" : "<DB>",
	"roles" : [
		{
			"role" : "readWrite",
			"db" : "< DB>"
		}
	]
```

**Comment by mrvautin** [comment]
Thanks. This has been fixed in 2cbb188c52f483ff9abc7a3e850518f417a20847.

Please keep raising any issues you come across.
